frontend/interfaces/software.ts

```diff
@@ -194,14 +196,19 @@ export const formatSoftwareType = ({
 /**
  * This list comprises all possible states of software install operations.
  */
+export const SOFTWARE_UNINSTALL_STATUSES = [
+  "uninstalled",
+  "pending_uninstall",
+  "failed_uninstall",
+] as const;
+
+export type SoftwareUninstallStatus = typeof SOFTWARE_UNINSTALL_STATUSES[number];
+
 export const SOFTWARE_INSTALL_STATUSES = [
-  "failed",
-  "failed_install",
   "installed",
-  "pending",
   "pending_install",
-  "pending_uninstall",
-  "failed_uninstall",
+  "failed_install",
+  ...SOFTWARE_UNINSTALL_STATUSES,
```

Comment:
If we have time for changes, I'd probably split this into three consts: install, uninstall, and a combined set of install and uninstall (which I'd just call SOFTWARE_STATUSES).

That way the composition is more clear: [...SOFTWARE_INSTALL_STATUSES, ...SOFTWARE_UNINSTALL_STATUSES].